src/Common/src/Interop/Unix/System.Net.Security.Native/Interop.GssApi.cs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static class GssApi
+    {
+        internal static bool EstablishSecurityContext(
+            ref SafeGssContextHandle context,
```

Comment:
CreateSecurityContext is not appropriate because the way this is going to be consumed is similar to the SSPI model in Windows. There will be several calls to InitializeSecurityContext (or Establish.. in our case) and the first call allocates the context. So either we need 2 methods or let the caller handle context creation. I think instead of changing the logic in this PR, we should wait till the PR for the consuming code is sent (expected in a few days). That would help in arriving at the right pattern for this.